app/main.py

```diff
@@ -1 +1,51 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None :
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None :
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str):
+        super().__init__(name)
+        for skill in ["JavaScript", "HTML", "CSS"]:
+            self.learn_skill(skill)
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(FrontendDeveloper):
```

Comment:
The `BackendDeveloper` class inherits from `FrontendDeveloper`, which might not be the intended design. Typically, `BackendDeveloper` and `FrontendDeveloper` should be separate classes, and `FullStackDeveloper` should inherit from both. Consider revising the class hierarchy to reflect this.